### PR TITLE
Quarkus kubernetes support for service binding

### PR DESCRIPTION
**Pull request by mjurc**
* Adding a test for Quarkus kubernetes support for service binding from
  Crunchy Postgres.

### Summary

(Summarize the problem solved by this PR, and how to verify it manually)

Please select the relevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Dependency update
- [ ] Refactoring
- [ ] Backport
- [x] New scenario (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### Checklist:
- [x] Methods and classes used in PR scenarios are meaningful
- [x] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

**Comment by pjgg** [comment]
Please review CI pipeline. Maybe you will have to enable the profile `service-binding-modules` on CI 

```
Could not find the selected project in the reactor: service-binding/postgresql-crunchy/
```

**Review by pjgg — CHANGES_REQUESTED** [review]
CI failures

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Inject
+    static OpenShiftClient ocClient;
+
+    @QuarkusApplication
+    static RestService app = new RestService()
+            .onPreStart(s -> createPostgresCluster());;
+
+    @AfterAll
+    public static void deletePostgresCluster() {
```
Comment (pjgg) [inline service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java:30]: And afterAll static method could have a more "generic name" something like "tearDown" or ..."afterAll" so if in the future we want to add more statements into this method, we don't have to change the method name.

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        deleteYaml("pg-cluster.yml");
+    }
+
+    /**
+     * This test verifies the application deploys successfully. If the binding fails, the application will not be
+     * available.
+     */
+    @Test
+    public void verify() {
```
Comment (pjgg) [inline service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java:39]: Verify is too poor method name

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        app.given()
+                .get()
```
Comment (pjgg) [inline service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java:41]: `get()` mmm interesting... I would expect something like ".get("/todo")"

```
  app.given()
                .get("/todo")
                .then()
                .statusCode(HttpStatus.SC_OK);
```

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        try {
+            new Command("oc", "-n", ocClient.project(), "wait", "--for", "condition=Ready", "--timeout=300s",
+                    "pods", "--all").runAndWait();
+        } catch (Exception e) {
+            deleteYaml("pg-cluster.yml");
+            Assertions.fail("PostgresCluster did not form correctly. Caused by: " + e.getMessage());
+        }
+    }
+
+    private static void applyYaml(String yamlFile) {
```
Comment (pjgg) [inline service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java:57]: `applyYaml` -> `applyCustomResourceDefinition` ?

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        try {
+            new Command("oc", "apply", "-f", Paths.get("target/test-classes/" + yamlFile).toString()).runAndWait();
+        } catch (Exception e) {
+            Assertions.fail("Failed to apply YAML file. Caused by: " + e.getMessage());
+        }
+    }
+
+    private static void deleteYaml(String yamlFile) {
```
Comment (pjgg) [inline service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java:65]: `deleteYaml` -> `deleteCustomResourceDefinition` ?

File: service-binding/postgresql-crunchy/src/test/resources/pg-cluster.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  image: registry.developers.crunchydata.com/crunchydata/crunchy-postgres:ubi8-14.2-1
+  postgresVersion: 14
+  instances:
+    - name: instance1
+      dataVolumeClaimSpec:
+        accessModes:
+          - "ReadWriteOnce"
+        resources:
+          requests:
+            storage: 1Gi
```
Comment (pjgg) [inline service-binding/postgresql-crunchy/src/test/resources/pg-cluster.yml:16]: `1Gi` -> `150Mi` ?

File: service-binding/postgresql-crunchy/src/test/resources/pg-cluster.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+      image: registry.developers.crunchydata.com/crunchydata/crunchy-pgbackrest:ubi8-2.38-0
+      repos:
+        - name: repo1
+          volume:
+            volumeClaimSpec:
+              accessModes:
+                - "ReadWriteOnce"
+              resources:
+                requests:
+                  storage: 1Gi
```
Comment (pjgg) [inline service-binding/postgresql-crunchy/src/test/resources/pg-cluster.yml:28]: `1Gi` -> `150Mi`

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+
+@OpenShiftScenario(deployment = OpenShiftDeploymentStrategy.UsingOpenShiftExtensionAndDockerBuildStrategy)
+public class OpenShiftPostgreSqlSbIT {
+
+    @Inject
+    static OpenShiftClient ocClient;
+
+    @QuarkusApplication
+    static RestService app = new RestService()
+            .onPreStart(s -> createPostgresCluster());;
```
Comment (pjgg) [inline service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java:27]: `;;` -> `;`

**Review by pjgg — CHANGES_REQUESTED** [review]

File: service-binding/postgresql-crunchy/src/main/docker/Dockerfile.jvm [hunk 9]
```diff
@@ -0,0 +1,94 @@
+####
```
[the remark this answers is not in the document]
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/main/docker/Dockerfile.jvm]: Updated.

File: service-binding/postgresql-crunchy/src/main/java/io/quarkus/ts/sb/postgresql/TodoResource.java [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    public Response create(Todo item) {
+        item.persist();
+        return Response.status(Status.CREATED).entity(item).build();
+    }
+
+    @GET
+    @Path("/{id}/complete")
+    @Transactional
+    public Response complete(@PathParam("id") Long id) {
+        Todo entity = Todo.findById(id);
```
[the remark this answers is not in the document]
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/main/java/io/quarkus/ts/sb/postgresql/TodoResource.java]: Updated.

File: service-binding/postgresql-crunchy/src/main/java/io/quarkus/ts/sb/postgresql/TodoResource.java [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+        entity.id = id;
```
[the remark this answers is not in the document]
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/main/java/io/quarkus/ts/sb/postgresql/TodoResource.java]: Updated.

File: service-binding/postgresql-crunchy/src/main/java/io/quarkus/ts/sb/postgresql/TodoResource.java [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+        entity.completed = true;
+        return Response.ok(entity).build();
+    }
+
+    @DELETE
+    @Transactional
+    @Path("/{id}")
+    public Response delete(@PathParam("id") Long id) {
+        Todo entity = Todo.findById(id);
```
[the remark this answers is not in the document]
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/main/java/io/quarkus/ts/sb/postgresql/TodoResource.java]: Updated.

File: service-binding/postgresql-crunchy/src/main/resources/application.properties [hunk 13]
```diff
@@ -0,0 +1,9 @@
+quarkus.kubernetes-service-binding.services.postgresql.api-version=postgres-operator.crunchydata.com/v1beta1
```
[the remark this answers is not in the document]
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/main/resources/application.properties:1]: Updated.

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java (hunk 4 above)
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java]: Updated.

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java (hunk 5 above)
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java]: Updated.

File: service-binding/postgresql-crunchy/src/test/resources/pg-cluster.yml (hunk 6 above)
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/test/resources/pg-cluster.yml]: Updated.

File: service-binding/postgresql-crunchy/src/test/resources/pg-cluster.yml (hunk 7 above)
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/test/resources/pg-cluster.yml]: Updated.

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java (hunk 8 above)
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java]: Updated.

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java (hunk 1 above)
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java]: Updated.

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java (hunk 2 above)
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java]: Updated.

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java [hunk 14]
```diff
@@ -0,0 +1,95 @@
...
+        return (outputString.contains("postgresoperator") && outputString.contains("service-binding-operator"));
+    }
+
+    private static void createPostgresCluster() {
+        try {
+            Thread.sleep(2000);
+        } catch (Exception e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        Assumptions.assumeTrue(areOperatorsInstalled());
```
Comment (mjurc) [inline service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java:67]: @pjgg I've also added an assumption about operators being installed on the cluster so we don't have to exclude this module in Maven executions.

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java [hunk 15]
```diff
@@ -0,0 +1,95 @@
...
+        } catch (Exception e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        String outputString = output.stream().collect(Collectors.joining(System.lineSeparator()));
+        return (outputString.contains("postgresoperator") && outputString.contains("service-binding-operator"));
+    }
+
+    private static void createPostgresCluster() {
+        try {
+            Thread.sleep(2000);
```
Comment (pjgg) [inline service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java:63]: Why is this Sleep thread necessary?

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java [hunk 16]
```diff
@@ -0,0 +1,95 @@
...
+        } catch (Exception e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        Assumptions.assumeTrue(areRequiredOperatorsInstalled());
+        applyCustomResourceDefinition("pg-cluster.yml");
+        try {
+            Thread.sleep(2000);
```
Comment (pjgg) [inline service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java:70]: and also this thread Sleep is really necessary?

**Review by pjgg — CHANGES_REQUESTED** [review]

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java (hunk 16 above)
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java:70]: Updated.

File: service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java (hunk 15 above)
Comment (mjurc) [reply service-binding/postgresql-crunchy/src/test/java/io/quarkus/ts/sb/postgresql/OpenShiftPostgreSqlSbIT.java]: Updated.

**Comment by pjgg** [comment]
Errors on CI are not related to this PR (is quota environmental error)
```
20:16:16,255 INFO  oc: error: build error: Failed to push image: error copying layers and metadata from ... received unexpected HTTP status: 500 Internal Server Error
```

**Review by pjgg — APPROVED** [review]
LGTM

**Pull request merged**